### PR TITLE
helm: Add Common label to all cilium resources

### PR DESCRIPTION
**Pull request by strongjz**
By adding a common label to all resources, this allows users to easily find all cilium deployed resources. This makes it easy to write OPA policies for all cilium resources. 

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [X] All code is covered by unit and/or runtime tests where feasible.
- [X] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [X] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [X] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

```release-note
Add a commonLabel to all cilium deployed resources
```

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
Comment (giorio94) [inline install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml]: Nit. Is there any reason why the format is not consistent across files?

File: install/kubernetes/cilium/templates/cilium-configmap.yaml
Comment (giorio94) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml]: The labels seem missing here.

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (giorio94) [inline install/kubernetes/cilium/values.yaml.tmpl]: All changes below should be reverted, as unrelated.

**Review by giorio94 — CHANGES_REQUESTED** [review]
Thanks! I've left a couple of comments inline. 

In addition:

* There are a bunch of workflows which failed, and need to be investigated (most likely to regenerate a few dependent files).
* Could you please rebase onto main to drop the merge commit? Merge commits are not allowed in this repo.
* Given that the expectation is to have these labels applied to all resources, I'd strongly suggest to add a linting step for that, as otherwise it is easy to forget about it. Something as simple as `grep -RL --exclude validate.yaml .Values.commonLabels install/kubernetes/cilium/templates/**/*.yaml` to be run as part of the helm smoke tests (`.github/workflows/tests-smoke.yaml`) could do the trick. For instance, it is already pointing to a few resources which seem to be legitimately missing the labels:
    ```
    install/kubernetes/cilium/templates/cilium-configmap.yaml
    install/kubernetes/cilium/templates/cilium-operator/rolebinding.yaml
    install/kubernetes/cilium/templates/cilium-operator/role.yaml
    ```

File: install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml
Comment (giorio94) [reply install/kubernetes/cilium/templates/cilium-agent/clusterrole.yaml]: This comment seems to still apply. A few differences between `{{- toYaml . | nindent 4 }}` and`{{- toYaml .Values.commonLabels | nindent 4 }}`, as well as indentation differences (whether that line is indented compared to the `with` line or not).

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (giorio94) [reply install/kubernetes/cilium/values.yaml.tmpl]: This still seems to apply.

File: .github/workflows/tests-smoke.yaml
Comment (giorio94) [inline .github/workflows/tests-smoke.yaml]: Thanks for adding this. The check should be extended a bit to provide user friendly errors in case of failure, rather than just the list of files. For instance checking the return code and echoing a bit of description alongside the list of failing files. One example could be [this one](https://github.com/cilium/cilium/blob/0b8beead6ba7a6e0c311f91baf9011355a34e183/.github/workflows/conformance-eks.yaml#L336-L340) for instance.

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (strongjz) [reply install/kubernetes/cilium/values.yaml.tmpl]: Hmm, had a couple resets. So yea, I'll fix it.

File: .github/workflows/tests-smoke.yaml
Comment (strongjz) [reply .github/workflows/tests-smoke.yaml]: I was thinking to add it make lint as well? So folks dont have to wait for CI to fail?

Comment (giorio94) [reply .github/workflows/tests-smoke.yaml]: Yeah, makes sense to me

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (strongjz) [inline install/kubernetes/cilium/values.yaml.tmpl]: i wonder if the linter or the make check fixed these? I didnt manually change these.

Comment (giorio94) [reply install/kubernetes/cilium/values.yaml.tmpl]: I think it is some linter configured in your editor, as the makefile should not be touching this file. I'd suggest reverting the changes in this file and then using a basic file editor to just add the common labels configuration. Sorry if I'm being picky about this, but this kind of white space changes increase the likelihood of hitting conflicts while backporting future changes, as well as mess up the git blame results.

File: .github/workflows/tests-smoke.yaml
Comment (giorio94) [inline .github/workflows/tests-smoke.yaml]: This is no longer needed given that it gets automatically executed by the make invocation below.

File: install/kubernetes/Makefile
Comment (giorio94) [inline install/kubernetes/Makefile]: This command is currently leaving behind the `missing_label` file when no missing labels are found.

**Review by giorio94 — CHANGES_REQUESTED** [review]
A couple of final comments inline, otherwise looks good to me. Thanks!

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (joestringer) [inline install/kubernetes/cilium/values.yaml.tmpl]: I'm curious, as I understand we already apply `app.kubernetes.io/part-of: cilium` for various resources. Is the intent here that the user has their own specific labels they want Cilium to have?

Comment (strongjz) [reply install/kubernetes/cilium/values.yaml.tmpl]: Yes, so they can apply specific OPA rules to cilium resources.

File: .github/workflows/tests-smoke.yaml
Comment (strongjz) [reply .github/workflows/tests-smoke.yaml]: so just change it to make -C install/kubernetes lint ? I dont see where it make lint for the helm chart is being done in CI?

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (strongjz) [reply install/kubernetes/cilium/values.yaml.tmpl]: fixed.

File: .github/workflows/tests-smoke.yaml
Comment (joestringer) [inline .github/workflows/tests-smoke.yaml]:
```suggestion
          
      - name: Run helm-charts
        run: |
          make -C install/kubernetes
```
No need for the changes in this file - running `make -C install/kubernetes` runs the `all` target there which includes the `lint` target.

**Comment by joestringer** [comment]
Looks like install currently fails with these changes: https://github.com/cilium/cilium/actions/runs/11618145788/job/32359194615?pr=35628#step:13:35

File: install/kubernetes/cilium/values.yaml.tmpl
Comment (giorio94) [inline install/kubernetes/cilium/values.yaml.tmpl]: Nit.

```suggestion
# -- commonLabels allows users to add common labels for all Cilium resources.
```

File: install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml
Comment (giorio94) [inline install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml]: This is required because you are already inside a loop, which redefines the root `.` object (it should be one of the causes of the CI failures).

```suggestion
    {{- with $.Values.commonLabels }}
```

File: install/kubernetes/cilium/templates/clustermesh-config/kvstoremesh-secret.yaml
Comment (giorio94) [inline install/kubernetes/cilium/templates/clustermesh-config/kvstoremesh-secret.yaml:12]: This hunk looks misplaced.

**Comment by giorio94** [comment]
Could you please run `make -C install/kubernetes` and commit the changes (CI will fail otherwise). Could you please also squash all commits into one?

**Comment by strongjz** [comment]
> make -C install/kubernetes

Squash just fine, Is this a mac sed issue?

```
 make -C install/kubernetes
: cilium/Chart.yaml
grep -lR -e 'version:' -e 'appVersion:' /Users/strongjz/go/src/github/cilium/cilium/install/kubernetes/cilium/ \
                | xargs -L 1 sed -i -e 's/''\([vV]ersion:\) ''[0-9]\+\.[0-9]\+\.[0-9]\+.*''/\1 '1.17.0-dev'/g'
sed -i 's;icon:.*;icon: https://cdn.jsdelivr.net/gh/cilium/cilium@main/Documentation/images/logo-solo.svg;' "/Users/strongjz/go/src/github/cilium/cilium/install/kubernetes/cilium/Chart.yaml"
sed: 1: "/Users/strongjz/go/src/ ...": bad flag in substitute command: 'a'
make: *** [update-chart] Error 1
```

**Comment by giorio94** [comment]
> > make -C install/kubernetes
> 
> Squash just fine, Is this a mac sed issue?
> 
> ```
>  make -C install/kubernetes
> : cilium/Chart.yaml
> grep -lR -e 'version:' -e 'appVersion:' /Users/strongjz/go/src/github/cilium/cilium/install/kubernetes/cilium/ \
>                 | xargs -L 1 sed -i -e 's/''\([vV]ersion:\) ''[0-9]\+\.[0-9]\+\.[0-9]\+.*''/\1 '1.17.0-dev'/g'
> sed -i 's;icon:.*;icon: https://cdn.jsdelivr.net/gh/cilium/cilium@main/Documentation/images/logo-solo.svg;' "/Users/strongjz/go/src/github/cilium/cilium/install/kubernetes/cilium/Chart.yaml"
> sed: 1: "/Users/strongjz/go/src/ ...": bad flag in substitute command: 'a'
> make: *** [update-chart] Error 1
> ```

Yep, looks like there's some issue with the `sed` version shipped there. I don't have a mac myself, but should be definitely fixed. Would you have time to file an issue to not forget about it? Anyways, for the moment I've just run the command locally and pushed the changes, to unblock this PR.

**Review by giorio94 — APPROVED** [review]

File: install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml:28]: Doesn't this also need the labels to be set on the `spec.template`?

File: install/kubernetes/cilium/templates/cilium-envoy/daemonset.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-envoy/daemonset.yaml:18]: Doesn't this also need the labels to be set on the `spec.template`?

File: install/kubernetes/cilium/templates/cilium-operator/deployment.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-operator/deployment.yaml:17]: Doesn't this also need the labels to be set on the `spec.template`?

File: install/kubernetes/cilium/templates/cilium-preflight/deployment.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/cilium-preflight/deployment.yaml:14]: Doesn't this also need the labels to be set on the `spec.template`?

File: install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml:18]: Doesn't this also need the labels to be set on the `spec.template`?

File: install/kubernetes/cilium/templates/hubble-relay/deployment.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/hubble-relay/deployment.yaml:15]: Doesn't this also need the labels to be set on the `spec.template`?

File: install/kubernetes/cilium/templates/hubble-ui/deployment.yaml
Comment (gandro) [inline install/kubernetes/cilium/templates/hubble-ui/deployment.yaml:15]: Doesn't this also need the labels to be set on the `spec.template`?

**Review by gandro — COMMENTED** [review]
Overall looks good to me. However, I wonder if we also want the actual pods created by Deployments and DaemonSets to also have the labels. I've highlighted the ones I found.

**Review by joestringer — APPROVED** [review]
:+1: for @cilium/docs-structure

File: install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml
Comment (gandro) [reply install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml:18]: This one is marked as resolve but I don't see the necessary change?

**Review by gandro — COMMENTED** [review]
Thanks for addressing my feedback. I think you missed one file ([install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml](https://github.com/cilium/cilium/pull/35628/files#diff-c85fb06a8dab414080c30e0cf3361b9baade1b357dd2d90e1a1797be2dbdcf31)), I unresolved the corresponding comment

**Review by gandro — APPROVED** [review]

**Comment by giorio94** [comment]
@strongjz There are a bunch of CI failures that are legitimate. Could you please take a look?

**Review by youngnick — APPROVED** [review]
LGTM

**Comment by strongjz** [comment]
can we marge this?

**Pull request merged**
